pt_miniscreen/core/components/list.py

```diff
@@ -91,43 +110,40 @@ def _scroll_transition(self):
         self._rows_snapshot = None
         self.state.update({"active_transition": None, "transition_progress": 0})
 
-    def scroll_up(self):
+    def scroll_to(self, direction, distance=1):
         if self.state["active_transition"] is not None:
-            logger.debug(f"{self} currently scrolling, ignoring scroll up")
+            logger.info(f"{self} currently scrolling, ignoring scroll")
             return
 
-        next_top_row_index = self.state["top_row_index"] - 1
-        if next_top_row_index < 0:
-            logger.debug(f"{self} has no more rows, ignoring scroll up")
-            return
-
-        Row = self.state["Rows"][next_top_row_index]
-        self.rows.insert(0, self.create_child(Row))
+        if direction == "UP":
+            if not self.can_scroll_up:
+                logger.info(f"{self} has no more rows, ignoring scroll up")
+                return
+
+            next_top_row_index = self.state["top_row_index"] - distance
+            Row = self.state["Rows"][next_top_row_index]
+            self.rows.insert(0, self.create_child(Row))
+        elif direction == "DOWN":
+            if not self.can_scroll_down:
+                logger.info(f"{self} has no more rows, ignoring scroll down")
+                return
+
+            next_top_row_index = self.state["top_row_index"] + distance
+            Row = self.state["Rows"][
+                next_top_row_index + self.state["num_visible_rows"] - 1
+            ]
+            self.rows.append(self.create_child(Row))
 
         self.state.update(
-            {"active_transition": "UP", "top_row_index": next_top_row_index}
+            {"active_transition": direction, "top_row_index": next_top_row_index}
         )
         threading.Thread(target=self._scroll_transition).start()
 
-    def scroll_down(self):
-        if self.state["active_transition"] is not None:
-            logger.debug(f"{self} currently scrolling, ignoring scroll down")
-            return
-
-        Rows = self.state["Rows"]
-        next_top_row_index = self.state["top_row_index"] + 1
-        max_top_row_index = len(Rows) - self.state["num_visible_rows"]
-        if next_top_row_index > max_top_row_index:
-            logger.debug(f"{self} has no more rows, ignoring scroll down")
-            return
-
-        Row = Rows[next_top_row_index + self.state["num_visible_rows"] - 1]
-        self.rows.append(self.create_child(Row))
+    def scroll_up(self):
```

Comment:
I think this can have distance in as an argument as well, seems like a good API to have!